Sources/Target/Common/ThreadBase.cpp

```diff
@@ -39,5 +39,36 @@ ErrorCode ThreadBase::beforeResume() {
 
   return kSuccess;
 }
+
+void ThreadBase::setBreakpointStopReason() {
+  auto *hwBpm = process()->hardwareBreakpointManager();
+  if (hwBpm == nullptr) {
+    return;
+  }
+
+  BreakpointManager::Site site;
+  int bpIdx = hwBpm->hit((Target::Thread *)this, site);
```

Comment:
Having to do a downcast here is lame. If you need to do a cast, use `static_cast`, but I think there's a way to structure this code without having to do any cast.